packages/tracing/src/browser/request.ts

```diff
@@ -133,17 +140,21 @@ export function instrumentOutgoingRequests(_options?: Partial<RequestInstrumenta
     };
   }
 
+  const shouldAttachHeaders = (url: string): boolean =>
+    tracingOrigins.some(origin => isMatchingPattern(url, origin)) ||
+    tracePropagationTargets.some(origin => isMatchingPattern(url, origin));
```

Comment:
I'm torn between loving the simplicity of this implementation and thinking we should include the kind of result-caching which is being pulled out of `shouldCreateSpan` in https://github.com/getsentry/sentry-javascript/pull/6039.

Realistically, in a modern browser on modern hardware, maybe it doesn't make enough of a difference to be worth the extra bytes. Opinions, @getsentry/team-web-sdk-frontend?

Reply:
as the person who put the cache setup there, it was a bit of a premature optimization.

I think it shouldn’t be that hard to add it back, so let’s just proceed with this.